Sources/OpenAISwift/OpenAISwift.swift

```diff
@@ -5,14 +5,39 @@ import FoundationXML
 #endif
 
 public enum OpenAIError: Error {
+    case internalError(error: InternalError)
     case genericError(error: Error)
     case decodingError(error: Error)
 }
 
+public struct RequestError: LocalizedError {
+    var errorDescription: String {
+        "Error creating the url"
+    }
+}
+
+public struct InternalError: LocalizedError {
+    let message: String
```

Comment:
```suggestion
    public let message: String
    public let type: String
    public let param: String?
    public let code: String?

    public var errorDescription: String {
         message
     }
```

Reply:
Make sure the properties are available from outside the OpenAISwift module, so apps and tools can show the proper `errorDescription` to the user.

Reply:
(Here is with the patch applied: https://github.com/adamrushy/OpenAISwift/pull/49)

Reply:
you are right, thanks!